mlir/lib/Dialect/Arith/IR/ArithCanonicalization.td

```diff
@@ -233,6 +233,50 @@ def CmpIExtUI :
             CPred<"$0.getValue() == arith::CmpIPredicate::eq || "
                   "$0.getValue() == arith::CmpIPredicate::ne">> $pred)]>;
 
+//===----------------------------------------------------------------------===//
+// SelectOp
+//===----------------------------------------------------------------------===//
+
+// select(not(pred), a, b) => select(pred, b, a)
+def SelectNotCond :
+    Pat<(SelectOp (Arith_XOrIOp $pred, (ConstantLikeMatcher APIntAttr:$ones)), $a, $b),
```

Comment:
`not` is expressed as `xor(pred, 1)`. Does this case also handle `xor(1, pred)`?

Reply:
My understanding is that commutative operators like `xor` are normalized so that constants always appear on the right hand side. I'll add a test to double-check.

Reply:
I've added an inverted call to the test case.